frontend/model/captureLogs.js

```diff
@@ -220,20 +164,41 @@ export function downloadLogs (elLink: Object): void {
   }
 }
 
-function clearLogs () {
-  let i = localStorage.length
-  while (i--) {
-    const key = localStorage.key(i)
-    if (typeof key === 'string' && key.indexOf('giConsole') >= 0) {
-      localStorage.removeItem(key)
+export function getLogger (): Object {
+  if (!logger) {
+    logger = createLogger(config)
+    const previousEntries = JSON.parse(getItem('entries'))
+
+    // If `maxEntries` is changed in a release, this will discard oldest logs as necessary.
+    if (config.maxEntries < previousEntries?.length) {
+      previousEntries.splice(0, previousEntries.length - config.maxEntries)
+    }
```

Comment:
Nice thinking, and very elegant design for the circular list. I almost thought this was a bug because it doesn't consider the offset, but then I realized it doesn't need to thanks to how you implemented `.toArray()`.

Reply:
I'm glad you like it! 🙂